src/cli_config.rs

```diff
@@ -228,16 +244,40 @@ pub fn cli_parse() -> CliConfig {
                 addr: addr.to_owned(),
             }
         })
-        .collect();
+        .collect::<Vec<_>>();
 
-    let threads_config = value_t!(matches.value_of("threads"), ThreadsConfig).unwrap_or_else(|e| e.exit());
+    let threads_config = if matches.is_present("threads") {
+        Some(value_t!(matches.value_of("threads"), ThreadsConfig).unwrap_or_else(|e| e.exit()))
+    } else {
+        None
+    };
 
-    let log_type = value_t!(matches.value_of("log-type"), LogType).unwrap_or_else(|e| e.exit());
+    let log_type = if matches.is_present("log-type") {
+        Some(value_t!(matches.value_of("log-type"), LogType).unwrap_or_else(|e| e.exit()))
+    } else {
+        None
+    };
 
-    let motd = value_t!(matches.value_of("motd"), String).unwrap_or_else(|e| e.exit());
+    let motd = matches.value_of("motd").map(|s| s.to_owned());
 
     let lan_discovery_enabled = !matches.is_present("no-lan");
 
+    let config_file = matches.value_of("config").map(|s| s.to_owned());
+
+    if config_file.is_some() && (
+        udp_addr.is_some() ||
```

Comment:
You can check in some other way. Like `if config.file.is_some && argc > 3`

Reply:
Maybe better to use a subcommand? (see my last comment)

Reply:
Yay! Good point, I like it.

Reply:
done